src/reports.cpp

```diff
@@ -527,7 +527,7 @@ static config unit_advancement_options(const unit* u)
 	if (!u) return config();
 	config res;
 	typedef std::pair<std::string, std::string> pair_string;
-	for (const pair_string &ps : u->advancement_icons()) {
+	for (const pair_string ps : u->advancement_icons()) {
```

Comment:
same here, assinga  const in the line above woudl be better than removing the `&`

Reply:
(alternativeley you can remove the line above altogether and just wrtie `for (const auto& ps : ...`)

Reply:
I'll use auto.  Using a typedef like that seems unnecessarily verbose.